nexus_constructor/json/load_from_json.py

```diff
@@ -28,12 +32,22 @@ def _retrieve_children_list(json_dict: dict) -> list:
         return []
 
 
+def _get_transformation_by_name(
+    component: Component, transformation_name: str
+) -> Transformation:
+    for transformation in component.transforms_list:
+        if transformation.name == transformation_name:
+            return transformation
+
+
 class JSONReader:
     def __init__(self, parent: QWidget):
         self.entry = Entry()
         self.entry.instrument = Instrument()
         self.parent = parent
         self.warnings = []
+        self.depends_on_paths = dict()
```

Comment:
Type hint and comment to explain what goes in these dictionaries would be really useful to aid understanding.
Here is an example for `self.depends_on_paths`, please do the same for `self.component_dictionary`.
```suggestion
        # key: component name, value: NeXus path pointing to transformation that component depends on
        self.depends_on_paths: Dict[str, str] = dict()
```

Reply:
@DolicaAkelloEgwel if you address this comment (document `self.component_dictionary`) then I think we could merge what you have done so far.